### ISSUE TITLE
Need easy way to format date without time

### ISSUE THREAD

**Issue by rowanseymour** [nyaruka/goflow#319]
Old engine has these in its context:

```
@date.tomorrow -> "03-07-2018"
@date.today -> "02-07-2018"
@date.yesterday -> "01-07-2018"
```

Currently there's no easy way to replicate that. We have `format_datetime`... and you can give it something like `@(format_datetime(now(), "DD-MM-YYYY"))` but we don't have access to the org date format in expressions. It could be `@(format_datetime(now(), "MM-DD-YYYY"))`

I would assume that outputting dates without time is sufficiently common that we want to make this super simple. Options off the top of my head...

```
@(format_datetime(now(), "DD-MM-YYYY")) // use org date format at *migrate* time
@(format_datetime(now(), env.date_format)) // add new env object to context
@(format_date(now())) // add new xfunction
```

I'm leaning toward the last option

**Comment by nicpottier** [comment]
Datetime formats aren't part of the environment? I thought that was.. how do we do the right thing with an empty `format_datetime` then? Or are you saying there isn't anything for a date but rather only datetime?

I find `format_date` slightly weird because you would then expect to be able to pass in formatting parameters, but not all of them would be valid (and in reality that function is then the same thing as `format_datetime` for all valid inputs).

That said, I like that better than the alternatives. We aren't bringing yesterday and tomorrow forward are we?

**Comment by rowanseymour** [comment]
Exactly - empty `format_datetime` outputs date+time. `tomorrow` and `yesterday` (which I don't think are used so much) are being migrated to `datetime_add(...)` calls.

Agree `format_date(...)` could cause confusion... what about "special formatting presets"...

```
@(format_datetime(now(), "d")
@(format_datetime(now(), "date")
```

**Comment by nicpottier** [comment]
I think I'd rather have a `format_date` which only honors datey things than have special secret parameters.

**Comment by rowanseymour** [comment]
Fixed in #322
